### ISSUE TITLE
Finish doesn't work from web control

### ISSUE THREAD

**Issue by DoctorPhilgood** [FoldingAtHome/fah-issues#1415]
If I tell FAH to "Finish up, then stop" from the web control, nothing happens, and this is the case for every computer I try this on. To get around this, I have to open up the advanced control and click "Finish" from there.

**Comment by Pepe-CH** [comment]
I can confirm this. I have the same issue. I'm using V7.6.9 / Win x64.

**Comment by Benica** [comment]
Same here. Checked Edge, Chrome, & Firefox, and all have the problem.

**Comment by PantherX** [comment]
Is this still an issue with the latest beta 7.6.12: https://foldingathome.org/beta/

**Comment by Benica** [comment]
> Is this still an issue with the latest beta 7.6.12: https://foldingathome.org/beta/

Yes

**Comment by anand-bhat** [comment]
From https://foldingforum.org/viewtopic.php?f=108&t=35084, "the button actually doesn't do anything at all besides close the popup. In comparison, the stop button has an additional line of code which immediately pauses folding, whereas there's nothing at all in the finish up button that suggests signalling the system to pause after a WU finishes."

```
$('#stop-popup .cancel').on('click', function(e) {
    e.preventDefault();
    $('#stop-popup').dialog('destroy');
});
```

**Comment by ejoepan1** [comment]
"Finish then stop" does not work for me.  F@H continues to run, pulls the next WU and continues to fold.  7.6.13 Win 10 x64  running CPU WU's only - no discrete GPU

**Comment by TunaErtemalp** [comment]
I had this problem on the Advanced Control today, too; not the WebControl. Last night I did FINISH on one machine, saw it in the FINISHING state, went to bed, woke up, the machine rebooted at 7am due to a scheduled reboot, about 4-5hrs after last evening's WU would have finished, and proceeded to grab a WU and restart working. Don't know if FINISHED+PAUSED state is supposed to persist over a reboot, but reporting it here. Even it isn't supposed to, it probably should.

**Comment by ipkh** [comment]
Finish doesn't stop any pending requests for work.  So if a slot is waiting on a download, requested a unit and us awaiting an available one, it will still download and start that unit.

**Comment by TunaErtemalp** [comment]
> Finish doesn't stop any pending requests for work. So if a slot is waiting on a download, requested a unit and us awaiting an available one, it will still download and start that unit.

Hmmm... That slot was a CPU slot, the only available slot on that machine, running a WU that was a few hours away from finishing, therefore around 60-70% complete by the time I clicked FINISH, and I have my "download next WU" at 99% completeness of the running WU, and there was no other WU waiting for its turn, a new WU would/should not have been downloaded. No?

**Comment by ipkh** [comment]
@TunaErtemalp In your case, unless you have pause-on-start set to true in expert settings, Folding starts at login/boot. The finish command only works for the current session.  A reboot starts a fresh session of folding.

**Comment by TunaErtemalp** [comment]
> @TunaErtemalp In your case, unless you have pause-on-start set to true in expert settings, Folding starts at login/boot. The finish command only works for the current session. A reboot starts a fresh session of folding.

Gotcha! I guess I thought it also meant Finish+Pause, especially since PAUSE of a slot persists across reboot. Learned something new! Sorry for the red herring spam... :)

**Comment by PantherX** [comment]
@TunaErtemalp 
If you paused a slot, it will remember the state upon reboot. If you finish a slot, it will not remember the state. AFAIK, this is a design implementation.

**Comment by bitnapper** [comment]
Doesn't work on Firefox/Mac either. Had to terminate the process to stop.

**Comment by anand-bhat** [comment]
This has been fixed in [7.6.21](https://foldingforum.org/viewtopic.php?f=0&t=36307&p=344916&hilit=finish#p344916).

**Comment by PantherX** [comment]
Hi All,

Closing this issue now since it has been fixed in the latest release 😄
